File: 0621-task-scheduler/0621-task-scheduler.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        vector<int> mp(26,0);
        int maxFreq=0, countMaxFreq=0, sz=tasks.size();
        for(char i: tasks){
            mp[i-'A']++;
            if ( mp[i-'A'] > maxFreq ){
                maxFreq=mp[i-'A'];
            }
        }
        for(int i=0;i<26;i++){
            if ( mp[i]==maxFreq ) countMaxFreq++;
        }
        int time = (maxFreq-1)*(n+1)+countMaxFreq;
        return max(time, sz);
    }
};

File: 0621-task-scheduler/0621-task-scheduler.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        l={}
        maxFreq=0
        countFreq=0
        sz=len(tasks)
        for i in tasks:
            l[i]=l.get(i,0)+1
        maxFreq=max(l.values())
        countFreq=sum( 1 for v in l.values() if v==maxFreq)
        return max(sz, (maxFreq-1)*(n+1)+countFreq)